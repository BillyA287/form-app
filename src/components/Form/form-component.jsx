import React from "react";
import UserDetails from '../UserDetails/user.details-component'
import PersonalDetails from '../Personal/personal-component'
import Confirm from '../Confirm/confirm-component'
import Success from '../Success/success-component'

export class Form extends React.Component {
  state = {
    step: 1,
    firstName: "",
    lastName: "",
    email: "",
    age: "",
    occupation: "",
    city: "",
    bio: ""
  };

  //next step
  nextStep = () => {
    const { step } = this.state;
    this.setState({
      step: step + 1,
    });
  };

  //previous step
  previousStep = () => {
    const { step } = this.state;
    this.setState({
      step: step - 1,
    });
  };

  //handle field change 
  handleChange = input => event =>{
this.setState({[input]: event.target.value});
  }

  render() {
      const {step} = this.state;
      const {firstName, lastName, email, age, occupation, city, bio} = this.state
      const values = {firstName, lastName, email, age, occupation, city, bio}
    

      switch(step){
          case 1:
              return (<UserDetails nextStep={this.nextStep} handleChange={this.handleChange} values={values}/>
                )
                case 2:
                    return (<PersonalDetails nextStep={this.nextStep} previousStep={this.previousStep} handleChange={this.handleChange} values={values} />)
                case 3:
                    return (
                      <Confirm
                        nextStep={this.nextStep}
                        previousStep={this.previousStep}
                        
                        values={values}
                      />
                    );
                case 4: return (<Success />)

                default :
              
      }
    
      
  }
}

export default Form