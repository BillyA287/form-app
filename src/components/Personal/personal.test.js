import React from 'react'
import {shallow} from 'enzyme'
import PersonalDetails from './personal-component'
import Form from '../Form/form-component'


it('Correctly renders PersonalDetails component', ()=> {
   expect(shallow(<PersonalDetails values={jest.fn()} handleChange={jest.fn()} />)).toMatchSnapshot()
})

const wrapper = shallow(<PersonalDetails handleChange={jest.fn()} values={jest.fn()}/>)




// expect(wrapper.find('[id="input"]').exists()).toBe(true);

wrapper.find('[id="input"]').simulate('change');
expect(wrapper.state()).toEqual({firstName: 'New value'});
