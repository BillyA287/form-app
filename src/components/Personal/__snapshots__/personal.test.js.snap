// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders PersonalDetails component 1`] = `ShallowWrapper {}`;
